packages/dds/map/src/interfaces.ts

```diff
@@ -234,8 +239,13 @@ export interface ISharedMapEvents extends ISharedObjectEvents {
     (event: "valueChanged", listener: (
         changed: IDirectoryValueChanged,
         local: boolean,
-        op: ISequencedDocumentMessage,
+        op: ISequencedDocumentMessage | null,
         target: IEventThisPlaceHolder) => void);
+    (event: "clear", listener: (
+        local: boolean,
+        op: ISequencedDocumentMessage | null,
+        target: IEventThisPlaceHolder
```

Comment:
Do we need the target arg?  Would probably prefer to omit unless we need it, to keep the signature simpler.  Both here and for other events.

Reply:
This was added explicitly in 2019 in this [commit](https://github.com/microsoft/FluidFramework/commit/c7d52fc091981449fd4e1c1c404c2d1cdeb69dee) 
@anthony-murphy raise the original issue (back in the old github repo)

Reply:
The gist of the it is that without it, it will be difficult to create a generic event handler (e.g. without closure at the time of listener registration)

Reply:
Nit: May be worth a comment in the code.

Reply:
I think this is more a general api design guideline.  Would be pretty tedious to put that on all event handers declaration?

Reply:
Ahh, I see. Agreed, a comment is the wrong place for such a note.